In [1]:
%matplotlib inline
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
import pysentani as sti

In [2]:
survey = pd.read_excel('../data-anonymous/sentani-merged-cleaned-anonymous-2014-11-29.xlsx')

In [3]:
# pysentani access type function
survey['access_type'] = sti.access_type(survey)

In [4]:
test = pd.Series([True, False, np.nan]).astype(bool)
test

0     True
1    False
2     True
dtype: bool

##fridge
* omitting access type to view raw data in each category

In [54]:
#this is the count for each response  in app_now fridge
now_fridge = survey[['app_now/fridge']].fillna('no response')
now_fridge['dummy'] = now_fridge['app_now/fridge'].astype(bool)
now_fridge.groupby(['app_now/fridge']).count().transpose()

app_now/fridge  0.0  1.0  no response
dummy           837  198          149

In [66]:
#this is the count for each response  in app_buy fridge
buy_fridge = survey[['app_buy/fridge']].fillna('no response')
buy_fridge['dummy'] = buy_fridge['app_buy/fridge'].astype(bool)
buy_fridge.groupby(['app_buy/fridge']).count().transpose()

app_buy/fridge  0.0  1.0  no response
dummy           427  371          386

The first two cells below show different combinations of responses. These both seem plausible because the amount of respondents wanting to buy has a max value of 371.

In [63]:
# app not owned = true(na's are originally true (~)then they become false)
# app want to buy= true, na's are false
fridge_var = ~survey['app_now/fridge'].astype(bool) & survey['app_buy/fridge'].fillna(0).astype(bool)
fridge_var.value_counts()

False    850
True     334
dtype: int64

In [64]:
# app not owned = true(na's treated as false become true)
# app want to buy = true (na's are false)
fridge_var1 = ~survey['app_now/fridge'].fillna(0).astype(bool) & survey['app_buy/fridge'].fillna(0).astype(bool)
fridge_var1.value_counts()

False    841
True     343
dtype: int64

549 > than 371 therfore impossible.

In [65]:
#if app not owned = true (na's ignored become false)
#if app want to buy = (true na's are also true)
fridge_var2 = ~survey['app_now/fridge'].astype(bool) & survey['app_buy/fridge'].astype(bool)
fridge_var2.value_counts()

False    635
True     549
dtype: int64

##Fan

In [58]:
now_fan = survey[['app_now/fan']].fillna('no response')
now_fan['dummy'] = now_fan['app_now/fan'].astype(bool)
now_fan.groupby(['app_now/fan']).count().transpose()

app_now/fan  0.0  1.0  no response
dummy        905  130          149

In [59]:
buy_fan= survey[['app_buy/fan']].fillna('no response')
buy_fan['dummy'] = buy_fan['app_buy/fan'].astype(bool)
buy_fan.groupby(['app_buy/fan']).count().transpose()

app_buy/fan  0.0  1.0  no response
dummy        647  151          386

##TV

In [13]:
now_TV = survey[['access_type', 'app_now/TV']].fillna('no response')
now_TV['dummy'] = now_TV['app_now/TV'].astype(bool)
now_TV.groupby(['access_type', 'app_now/TV']).count().transpose()

access_type PLN_grid                  PLN_microgrid                   \
app_now/TV       0.0  1.0 no response           0.0  1.0 no response   
dummy            105  463          51            30  128          12   

access_type community_microgrid                 no_access                   
app_now/TV                  0.0 1.0 no response       0.0  1.0 no response  
dummy                         4  48           2        52  205          84

In [7]:
buy_TV = survey[['access_type', 'app_buy/TV']].fillna('no response')
buy_TV['dummy'] = buy_TV['app_buy/TV'].astype(bool)
buy_TV.groupby(['access_type', 'app_buy/TV']).count().transpose()

access_type PLN_grid                 PLN_microgrid                  \
app_buy/TV       0.0 1.0 no response           0.0 1.0 no response   
dummy            339  87         193            99  18          53   

access_type community_microgrid                 no_access                  
app_buy/TV                  0.0 1.0 no response       0.0 1.0 no response  
dummy                        30  16           8       146  63         132

##radio

In [14]:
now_radio = survey[['access_type', 'app_now/radio']].fillna('no response')
now_radio['dummy'] = now_radio['app_now/radio'].astype(bool)
now_radio.groupby(['access_type', 'app_now/radio']).count().transpose()

access_type   PLN_grid                  PLN_microgrid                  \
app_now/radio      0.0  1.0 no response           0.0 1.0 no response   
dummy              397  171          51           114  44          12   

access_type   community_microgrid                 no_access                   
app_now/radio                 0.0 1.0 no response       0.0  1.0 no response  
dummy                          26  26           2       146  111          84

In [8]:
buy_radio = survey[['access_type', 'app_buy/radio']].fillna('no response')
buy_radio['dummy'] = buy_radio['app_buy/radio'].astype(bool)
buy_radio.groupby(['access_type', 'app_buy/radio']).count().transpose()

access_type   PLN_grid                 PLN_microgrid                  \
app_buy/radio      0.0 1.0 no response           0.0 1.0 no response   
dummy              395  31         193           110   7          53   

access_type   community_microgrid                 no_access                  
app_buy/radio                 0.0 1.0 no response       0.0 1.0 no response  
dummy                          32  14           8       159  50         132

##rice_cooker

In [15]:
now_rc = survey[['access_type', 'app_now/rice_cooker']].fillna('no response')
now_rc['dummy'] = now_rc['app_now/rice_cooker'].astype(bool)
now_rc.groupby(['access_type', 'app_now/rice_cooker']).count().transpose()

access_type         PLN_grid                  PLN_microgrid                  \
app_now/rice_cooker      0.0  1.0 no response           0.0 1.0 no response   
dummy                    411  157          51           156   2          12   

access_type         community_microgrid                 no_access      \
app_now/rice_cooker                 0.0 1.0 no response       0.0 1.0   
dummy                                45   7           2       249   8   

access_type                      
app_now/rice_cooker no response  
dummy                        84

In [9]:
buy_rc = survey[['access_type', 'app_buy/rice_cooker']].fillna('no response')
buy_rc['dummy'] = buy_rc['app_buy/rice_cooker'].astype(bool)
buy_rc.groupby(['access_type', 'app_buy/rice_cooker']).count().transpose()

access_type         PLN_grid                  PLN_microgrid                  \
app_buy/rice_cooker      0.0  1.0 no response           0.0 1.0 no response   
dummy                    326  100         193            95  22          53   

access_type         community_microgrid                 no_access      \
app_buy/rice_cooker                 0.0 1.0 no response       0.0 1.0   
dummy                                29  17           8       156  53   

access_type                      
app_buy/rice_cooker no response  
dummy                       132

##lighting

In [16]:
now_light = survey[['access_type', 'app_now/lighting']].fillna('no response')
now_light['dummy'] = now_light['app_now/lighting'].astype(bool)
now_light.groupby(['access_type', 'app_now/lighting']).count().transpose()

access_type      PLN_grid                  PLN_microgrid                   \
app_now/lighting      0.0  1.0 no response           0.0  1.0 no response   
dummy                  39  529          51             8  150          12   

access_type      community_microgrid                 no_access       \
app_now/lighting                 0.0 1.0 no response       0.0  1.0   
dummy                              7  45           2        30  227   

access_type                   
app_now/lighting no response  
dummy                     84

In [10]:
buy_light = survey[['access_type', 'app_buy/lighting']].fillna('no response')
buy_light['dummy'] = buy_light['app_buy/lighting'].astype(bool)
buy_light.groupby(['access_type', 'app_buy/lighting']).count().transpose()

access_type      PLN_grid                 PLN_microgrid                  \
app_buy/lighting      0.0 1.0 no response           0.0 1.0 no response   
dummy                 364  62         193           116   1          53   

access_type      community_microgrid                 no_access       \
app_buy/lighting                 0.0 1.0 no response       0.0  1.0   
dummy                             13  33           8        82  127   

access_type                   
app_buy/lighting no response  
dummy                    132

In [27]:

#df = survey[[('app_now/fan'),'app_buy/fan']]
#(df[(df['app_now/fan'] ==0) & (df['app_buy/fan']>0)]).count()

In [28]:
#df1 = survey[['app_now/fan','app_buy/fan']]
#(df1[(df1['app_now/fan'] ==np.nan) & (df1['app_buy/fan']>0)]).count()

In [19]:
appliance_list = ['fridge', 'rice_cooker', 'fan', 'radio', 'TV', 'lighting' ]

###NaN's as false??

In [60]:
"""def calc_willingness(survey, appliance):
    app_now_appliance = 'app_now/' + appliance
    app_buy_appliance = 'app_buy/' + appliance
    survey[appliance] = ~survey[app_now_appliance].astype(bool)& survey[app_buy_appliance].fillna(0).astype(bool)
    

WBlist = []
for appliance in appliance_list:
    calc_willingness(survey, appliance)
    WBlist.append(appliance)
    
subsurvey = survey[['access_type'] + WBlist]
subsurvey"""

"def calc_willingness(survey, appliance):\n    app_now_appliance = 'app_now/' + appliance\n    app_buy_appliance = 'app_buy/' + appliance\n    survey[appliance] = ~survey[app_now_appliance].astype(bool)& survey[app_buy_appliance].fillna(0).astype(bool)\n    \n\nWBlist = []\nfor appliance in appliance_list:\n    calc_willingness(survey, appliance)\n    WBlist.append(appliance)\n    \nsubsurvey = survey[['access_type'] + WBlist]\nsubsurvey"

In [21]:
"""analyzed = subsurvey.groupby('access_type').agg(['mean','count'])
#analyzed['access_type'] = analyzed.index 
analyzed"""

"analyzed = subsurvey.groupby('access_type').agg(['mean','count'])\n#analyzed['access_type'] = analyzed.index \nanalyzed"

###NaN's as true

In [61]:
"""def calc_willingness1(survey, appliance):
    app_now_appliance = 'app_now/' + appliance
    app_buy_appliance = 'app_buy/' + appliance
    survey[appliance] = ~survey[app_now_appliance].astype(bool)& survey[app_buy_appliance].astype(bool)

WBlist1 = []
for appliance in appliance_list:
    calc_willingness1(survey, appliance)
    WBlist1.append(appliance)
    
subsurvey1 = survey[['access_type'] + WBlist1]
subsurvey1"""

"def calc_willingness1(survey, appliance):\n    app_now_appliance = 'app_now/' + appliance\n    app_buy_appliance = 'app_buy/' + appliance\n    survey[appliance] = ~survey[app_now_appliance].astype(bool)& survey[app_buy_appliance].astype(bool)\n\nWBlist1 = []\nfor appliance in appliance_list:\n    calc_willingness1(survey, appliance)\n    WBlist1.append(appliance)\n    \nsubsurvey1 = survey[['access_type'] + WBlist1]\nsubsurvey1"

In [23]:
"""analyzed1 = subsurvey1.groupby('access_type').count()
analyzed1['access_type'] = analyzed1.index 
analyzed1"""

"analyzed1 = subsurvey1.groupby('access_type').count()\nanalyzed1['access_type'] = analyzed1.index \nanalyzed1"